src/pe.cc

```diff
@@ -109,23 +110,37 @@ bool PeFile::Initialize() {
     return false;
   }
 
-  string_view exe_header =
-      GetRegion(dos_header_.AddressOfNewExeHeader, sizeof(pe_header));
-  memcpy(&pe_header_, exe_header.data(), exe_header.size());
+  PE_TYPE Magic;
+  auto pe_end =
+      CheckedAdd(dos_header_.AddressOfNewExeHeader, sizeof(pe_header_));
+  if (CheckedAdd(pe_end, sizeof(Magic)) > data_.size()) {
+    // Cannot fit the headers / magic from optional header
+    return false;
+  }
+
+  memcpy(&pe_header_, data_.data() + dos_header_.AddressOfNewExeHeader,
+         sizeof(pe_header_));
 
   if (!std::equal(pe_header_.signature, pe_header_.signature + sizeof(PE_Magic),
                   std::begin(PE_Magic))) {
     // Not a PE file.
     return false;
   }
 
-  // TODO(mj): Parse PE header further to determine this
-  is_64bit_ = false;
+  memcpy(&Magic, data_.data() + pe_end, sizeof(Magic));
+
+  if (Magic != PE_TYPE::PE32 && Magic != PE_TYPE::PE32_PLUS) {
+    // Unknown PE magic
+    return false;
+  }
+
+  is_64bit_ = Magic == PE_TYPE::PE32_PLUS;
```

Comment:
Some structs are different on 64 bit (none of those are used yet, but I wanted to add this check in advance)